lib/puppet/pops/pcore.rb

```diff
@@ -2,14 +2,32 @@
 
 module Puppet::Pops
 module Pcore
+  TYPE_URI_RX = Types::TypeFactory.regexp(URI.regexp)
+  TYPE_URI = Types::TypeFactory.pattern(TYPE_URI_RX)
+  TYPE_SIMPLE_TYPE_NAME = Types::TypeFactory.pattern(/[A-Z]\w*/)
   TYPE_QUALIFIED_REFERENCE = Types::TypeFactory.pattern(Types::TypeFactory.regexp(Patterns::CLASSREF_EXT))
 
+  KEY_PCORE_URI = 'pcore_uri'.freeze
+  KEY_PCORE_VERSION = 'pcore_version'.freeze
+
+  PCORE_URI = 'http://puppet.com/2016.1/pcore'
+  PCORE_VERSION = Semantic::Version.new(1,0,0)
+  PARSABLE_PCORE_VERSIONS = Semantic::VersionRange.parse('1.x')
+
   RUNTIME_NAME_AUTHORITY = 'http://puppet.com/2016.1/runtime'
 
   def self.init(loader, ir)
-    add_alias('Puppet::Pcore::QualifiedReference', TYPE_QUALIFIED_REFERENCE, loader)
+    add_alias('Pcore::URI_RX', TYPE_URI_RX, loader)
+    add_alias('Pcore::URI', TYPE_URI, loader)
```

Comment:
Shouldn't it take `Initializer[URI]` which would be a variant of the signatures to URI new? Agree it is a separate ticket, and so is Initializer functionality.